### PR TITLE
[skip-ci] Add missing xml docs for ExceptionHandler

### LINKED ISSUE #48640: IExceptionHandler is missing XML docs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

See https://github.com/dotnet/aspnetcore/blob/e4c6621d9a09cae8565ad918ba5083c9b28840bd/src/Middleware/Diagnostics/src/ExceptionHandler/IExceptionHandler.cs

### Expected Behavior

Has useful XML docs.

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### .NET Version

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by divyeshio**
## Description

Adds missing inline xml docs for IExceptionHandler and AddExceptionHandler method

Fixes #48640

### CODE REVIEW

**Review by amcasey — APPROVED** [review]

File: src/Middleware/Diagnostics/src/ExceptionHandler/IExceptionHandler.cs [hunk 1]
```diff
@@ -6,7 +6,9 @@
 namespace Microsoft.AspNetCore.Diagnostics;
 
 /// <summary>
-/// 
+/// Represents an interface for handling exceptions in ASP.NET Core applications.
+/// Implementations of this interface can provide custom exception handling logic for
+/// different scenarios in the application.
 /// </summary>
 public interface IExceptionHandler
```
Comment (adityamandaleeka) [inline src/Middleware/Diagnostics/src/ExceptionHandler/IExceptionHandler.cs:13]: `TryHandleAsync` below also needs to be filled in.

Comment (adityamandaleeka) [reply src/Middleware/Diagnostics/src/ExceptionHandler/IExceptionHandler.cs:13]: Other than that, this looks good to me. Thanks for doing it.

File: src/Middleware/Diagnostics/src/ExceptionHandler/IExceptionHandler.cs [hunk 2]
```diff
@@ -6,7 +6,9 @@
 namespace Microsoft.AspNetCore.Diagnostics;
 
 /// <summary>
-/// 
+/// Represents an interface for handling exceptions in ASP.NET Core applications.
+/// Implementations of this interface can provide custom exception handling logic for
+/// different scenarios in the application.
```
Comment (JamesNK) [inline src/Middleware/Diagnostics/src/ExceptionHandler/IExceptionHandler.cs:11]:
```suggestion
/// Represents an interface for handling exceptions in ASP.NET Core applications.
/// `IExceptionHandler` implementations are used by the exception handler middleware.
```

**Review by JamesNK — DISMISSED** [review]

**Review by Tratcher — APPROVED** [review]
Thanks

**Pull request merged**